netdcli/netdcli.go

```diff
@@ -373,46 +385,43 @@ func executeOpts(opts *cliOpts) error {
 
 	err := validateOpts(opts)
 	if err != nil {
-		log.Fatalf("error %v validating opts\n", err)
+		return err
 	}
 
-	etcdDriver := &state.EtcdStateDriver{}
-	driverConfig := &state.EtcdStateDriverConfig{}
-	driverConfig.Etcd.Machines = []string{opts.etcdURL}
-	config := &core.Config{V: driverConfig}
-	err = etcdDriver.Init(config)
+	stateDriver, err := utils.GetStateDriver()
```

Comment:
Got it, yes that will work. Let me make that change, for now I will expect the string to be url and process it to a json string as expected by the constructor.

I think the current change is more transient as netdcli/netmaster are expected to quickly evolve to work across different state and network-drivers and have more complex configurations. Once we have that we can start taking a json configuration as argument itself like we do for netplugin.